File: app/src/main/java/co/deucate/smsbomber/LogAdapter.java
package co.deucate.smsbomber;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class LogAdapter extends RecyclerView.Adapter<LogAdapter.ViewHolder> {

    private List<String> mList = new ArrayList<>();

    @NonNull
    @Override
    public LogAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recyclerview_text, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull LogAdapter.ViewHolder holder, int position) {
        String log = mList.get(position);
        holder.mTextView.setText(log);
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView mTextView;

        ViewHolder(View itemView) {
            super(itemView);

            mTextView = itemView.findViewById(R.id.recyclerTV);

        }
    }
}

File: app/src/main/java/co/deucate/smsbomber/LogAdapter.kt
package co.deucate.smsbomber

import android.graphics.Color
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList

class LogAdapter(private var logs: ArrayList<String>) : RecyclerView.Adapter<LogAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LogAdapter.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.recyclerview_text, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: LogAdapter.ViewHolder, position: Int) {
        var log = logs[position]
        val firstThree = log[0].toString()+log[1].toString()+log[2].toString()
        val finalString: String

        when (firstThree) {
            "Err" -> {
                log = log.replace("Err","")
                log = log.replace("orcode","Errorcode")
                finalString = "> $log"
                holder.mTextView.setTextColor(Color.RED)
            }
            "???" -> {
                log = log.replace("???","")
                finalString = "> $log"
                holder.mTextView.setTextColor(Color.YELLOW)
            }
            else -> finalString = log
        }

        holder.mTextView.text = finalString
    }

    override fun getItemCount(): Int {
        return logs.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var mTextView: TextView = itemView.findViewById(R.id.recyclerTV)

    }
}
